### ISSUE TITLE
getStats error when using Chrome M54 canary

### ISSUE THREAD

**Issue by juberti** [webrtc/apprtc#340]
Every getStats call results in "Uncaught (in promise) TypeError: Failed to execute 'getStats' on 'RTCPeerConnection': parameter 2 is not of type 'MediaStreamTrack'.(…)"

Using Chrome Version 54.0.2815.0 canary (64-bit) on Mac OS X

**Comment by juberti** [comment]
@alvestrand, any idea what changed here?

**Comment by alvestrand** [comment]
This is this week's canary with "all experiments on"?
Someone's feature-detecting on the presence of the standard getStats signature. The signature's been added behind a flag, but there's no code there yet.

The flag is set to "RTCPeerConnectionNewGetStats status=test", so it SHOULD NOT have been visible to a browser, even with "all experiments on".

**Comment by alvestrand** [comment]
On second reading, the error message is wrong for that. What does the caller look like?

**Comment by KaptenJansson** [comment]
It's a change made by hbos@ which basically fixes a bug we have had for a long time (i.e. stricter type checking on the arguments passed in to getStats) however in the bug he mentions it should not break existing apps which it did. https://bugs.chromium.org/p/chromium/issues/detail?id=632457.

Will fix on our side as well.

**Comment by henbos** [comment]
When I overloaded getStats I had to change the signature of the original getStats, e.g. change from void to Promise<void>, whether or not RTCPeerConnectionNewGetStats is on or off. I made a mistake by thinking I should have dropped the LegacyInterfaceTypeChecking label of the original getStats as well. This what is causing this issue. (We're trying to call it with an error callback but our getStats doesn't and never did have a function that takes an errror callback, LegacyInterfaceTypeChecking made it possible to call it like that anyway by defaulting that argument to null, making it a valid call.) I'll resolve the issue by re-adding LegacyInterfaceTypeChecking. crbug.com/632457
